src/main/java/bio/terra/cli/service/SamService.java

```diff
@@ -98,22 +99,11 @@ public SystemStatus getStatus() {
    * This is not the same thing as registering a user.
    *
    * @param userEmail email to invite
-   * @return SAM object with details about the user
    */
-  public UserStatusDetails inviteUser(String userEmail) {
-    UsersApi samUsersApi = new UsersApi(apiClient);
-    try {
-      logger.info("Inviting new user: {}", userEmail);
-      UserStatusDetails userStatusDetails =
-          HttpUtils.callWithRetries(
-              () -> samUsersApi.inviteUser(userEmail), SamService::isRetryable);
-      logger.info("Invited new user: {}", userStatusDetails);
-      return userStatusDetails;
-    } catch (ApiException | InterruptedException ex) {
-      throw new SystemException("Error inviting user in SAM.", ex);
-    } finally {
-      closeConnectionPool();
-    }
+  public void inviteUser(String userEmail) throws ApiException {
+    logger.info("Inviting new user: {}", userEmail);
+    UserStatusDetails userStatusDetails = new UsersApi(apiClient).inviteUser(userEmail);
```

Comment:
up to you: It's still a odd to me that this is the only SamService function that doesn't do its own retrying. I don't think the fact that this is frequently (only?) used with handleOneTimeError to be a compelling reason for the assymetry. I think it'd be preferablle to not retry the SamService.inviteUser (`x -> false`) when it's used in handleOneTimeError and rely on SamService to do the retry. I think then we also don't need to expose SamService::isRetryable and let Sam retry logic be totally encapsulated by the SamService.

Reply:
I see your point. I slightly prefer not nesting retries, because I think it makes the code less readable. Going to leave as is for now, but open to changing later if this gets too verbose.